M2/Macaulay2/packages/NoetherianOperators.m2

```diff
@@ -0,0 +1,3649 @@
+    if ring first L =!= R then error "expected Noetherian operators and prime in same ring";
+    indVars := support first independentSets P;
+    FF := frac(R/P);
+    S := FF monoid R;
+
+    mapDiff := map(S,R, vars S);
+    mapCoef := map(coefficientRing S, R);
+    V := L / (op -> applyPairs(op, (a,b) -> (mapDiff a, promote(mapCoef b,S))));
+
+    I := vectorAnn(V);
+    R' := R monoid R;
+    coefs := liftColumnsPunctualHilbert(lift(last coefficients gens I, coefficientRing S), R);
+    mons := (map(R', S, vars R'))(first coefficients gens I);
+    I' := ideal(mons * coefs);
+
+    X := R'/(I'+P);
+    Lmap := apply(numgens R, i -> R_i => promote(R_i, R') + R'_i);
+    mapRtoX := map(X, R, Lmap);
+    Q := ker mapRtoX;
+    for v in indVars do -- heuristic for faster computation 
+      Q = saturate(Q, ideal(v));
+    first select(primaryDecomposition(Q), K -> radical(K) == P)    
+)
+
+TEST ///
+debug NoetherianOperators
+R = QQ[x,y,t]
+I = ideal(x^2, y^2 - t*x)
+L = getNoetherianOperatorsHilb(I)
+P = radical I
+Q = getIdealFromNoetherianOperators(L,P)
+assert(Q == I)
+///
+----------------------------------------------------------
+
+undocumented {
+    (expression, DiffOp),
+    (expression, InterpolatedDiffOp),
+    (net, DiffOp),
+    (net, InterpolatedDiffOp),
+    (toExternalString, ZeroDiffOp),
+    (toString, DiffOp)
+}
+
+beginDocumentation()
+
+doc ///
+     Key
+       NoetherianOperators
+     Headline
+       numerically compute local dual space and Hilbert functions
+     Description
+       Text
+	   The @EM "NoetherianOperators"@ package includes algorithms for computing Noetherian operators and local dual 
+	   spaces of polynomial ideals, and related local combinatorial data about its scheme structure.
+    	    	
+	   The problem of characterizing ideal membership with differential conditions was first addressed by 
+	   Gr\"obner ("Uber eine neue idealtheoretische Grundlegung der algebraischen Geometrie", Math. Ann. 115 (1938), no. 1, 333–358).
+	   Despite this early algebraic interest by Gr\"obner, a complete description of primary ideals in terms of differential operators was first obtained by analysts in the Fundamental Principle of Ehrenpreis and Palamodov.
+	   At the core of the Fundamental Principle, one has the notion of Noetherian operators to describe a primary ideal. 
+	   
+	   In case of an ideal supported at one point a set of Noetherian operators forms a Macaulay inverse system that spans the dual space of the ideal. 
+	   These notions relate to the work of 
+	   Macaulay ("The algebraic theory of modular systems", Cambridge Press, (1916)). 
+	    
+	   In this package, we implement several (exact symbolic and approximate numerical) algorithms for the computation of a set of Noetherian operators describing a primary ideal. 
+    	           	   
+           Methods and types for computing and manipulating Noetherian operators:
+
+           @UL {
+               {TO noetherianOperators, UL {
```

Comment:
How does this list look in the html form of the documentation?  I'm suspicious, because the items in a list should each be inside an `LI` item.

Reply:
At least on my computer, it seems to look OK (ignore the dark mode theme): 

![noethOps_doc](https://user-images.githubusercontent.com/12538422/100375758-4ca31980-2fdc-11eb-91fc-5e8795817eac.png)

Reply:
By the way, you can also use the "Tree" keyword of SimpleDoc for this purpose.

Reply:
Okay, it seems UL is eager to wrap items inside LI, so there is no problem.